SerialPrograms/Source/PokemonHome/Programs/PokemonHome_BoxSorting.cpp

```diff
@@ -642,12 +665,51 @@ void BoxSorting::program(SingleSwitchProgramEnvironment& env, SwitchControllerCo
                         }
                         boxes_data[get_index(box_nb, row, column)]->ot_id = ot_id;
 
-                        // NOTE edit when adding new struct members (detections go here likely)
-
                         // level_box
-                        // ot_box
+                        int level = OCR::read_number_waterfill(env.console, extract_box_reference(screen, level_box), 0xff000000, 0xff7f7f7f);
+                        if (level < 0 || level > 100) {
+                            dump_image(env.console, ProgramInfo(), "ReadSummary_Level", screen);
+                        }
+                        boxes_data[get_index(box_nb, row, column)]->level = (uint16_t)level;
+                        env.console.log("Level: " + std::to_string(level), COLOR_GREEN);
+
+                        // language
+                        std::string language = OCR::ocr_read(Language::English, extract_box_reference(screen, language_box));

```

Comment:
Raw OCR isn't perfect and it'd be better to have a dictionnary (a list of words it can find, in this case all languages). That way, OCR will be able to correct minor errors while reading. Also check that all languages slugs are written in latin character (otherwise, the language need to be adapted and can't be english all the time)